webapp/tests/forms/test_fields.py

```diff
@@ -193,4 +194,22 @@ def test_if_data_given_and_valid_formdata_then_value_equals_form_data_as_is(self
         self.form.process(formdata=MultiDict({'date_field': correct_date_input}),
                           data={'date_field': dt.date(1979, 9, 19)})
         self.form.validate()
-        self.assertEqual([31, 7, 1980], self.form.date_field._value())
\ No newline at end of file
+        self.assertEqual([31, 7, 1980], self.form.date_field._value())
+
+
+class UnlockCodeForm(SteuerlotseBaseForm):
+    unlock_code = UnlockCodeField()
+
+
+class TestUnlockCodeFieldValue(unittest.TestCase):
+
+    def setUp(self):
+        self.form = UnlockCodeForm()
+
+    def test_if_data_given_and_lowercase_then_value_equals_uppercase_data(self):
+        """ Simulates a POST request with valid formdata and no prefilled data."""
+        unlock_input = ['aaaa', '12ade', '1l2ö']
+        self.form.process(formdata=MultiDict({'unlock_code': unlock_input}))
+        self.form.validate()
+        self.assertEqual(['AAAA', '12ADE', '1L2Ö'], self.form.unlock_code._value())
+
```

Comment:
Is creating a form necessary for testing the field behaviour?

Reply:
It is not necessary, we could also test `process_formdata`. However, I think it is better that way because in the end we only care about the result of `form.process` and `form.validate`. Having it set up that way we can ensure that the field is set up in the same way as it is in a form and that we've changed the correct method of the field. It is also testing the behavior in the same way that we are going to use it.